cli/lsp/tsc.rs

```diff
@@ -480,6 +481,59 @@ impl DocumentSpan {
   }
 }
 
+#[derive(Debug, Clone, Deserialize)]
```

Comment:
Does it need to be Clone?

Reply:
Yes, because like `LineIndex`, when we provide it from the document cache to be operated on, we provide a clone, as providing a reference gets complex pointer lifetimes and we can't move it out.